include/seqan3/range/views/minimiser.hpp

```diff
@@ -555,6 +556,10 @@ struct minimiser_fn
                       "The range1 parameter to views::minimiser must model std::ranges::forward_range.");
         static_assert(std::ranges::forward_range<urng2_t>,
                       "The range2 parameter to views::minimiser must model std::ranges::forward_range.");
+        static_assert(std::ranges::sized_range<urng1_t>,
+                      "The range1 parameter to views::minimiser must model std::ranges::sized_range.");
+        static_assert(std::ranges::sized_range<urng2_t>,
+                      "The range2 parameter to views::minimiser must model std::ranges::sized_range.");
 
         if (std::ranges::size(urange1) != std::ranges::size(urange2))
```

Comment:
On a second thought, we could also use distance which also gives the size back, but has no constant time guarantee.

```suggestion
        if (std::ranges::distance(urange1) != std::ranges::distance(urange2))
```

Reply:
That is now a little weird. In the adaptor defintion I see for the first time the sized_range constrained? Is that really so? It was never mentioned before and also the code changes so far do not indicate that sizedness is needed. In that case @marehr is right and the only generic solution would be to use distance. However, it feels more appropriate to put that check into the constructor of the class, since the adaptor is just delegating to the constructor. Hence, I could create an invalid class itself, which is not good.

Reply:
The reason why sized_range is needed is that for some reason for the test with `take until | reverse` is wrongly returning that it is a sized range and therefore initializing the kmer_hash size function, which then leads into an error message. So therefore the static assert is needed at the moment...
I will use distance and at the check  into the minimiser class. :smile: